addons/pos_razorpay/models/pos_payment_method.py

```diff
@@ -0,0 +1,123 @@
+
+    razorpay_tid = fields.Char(string='Razorpay Device Serial No', help="Device Serial No \n ex: 7000012300")
+    razorpay_allowed_payment_modes = fields.Selection(selection=[('all', 'All'), ('card', 'Card'), ('upi', 'UPI'), ('bharatqr', 'BHARATQR')], default='all', help="Choose allow payment mode: \n All/Card/UPI or QR")
+    razorpay_username = fields.Char(string="Razorpay Username", help="Username(Device Login) \n ex: 1234500121")
+    razorpay_api_key = fields.Char(string="Razorpay API Key", help="API Key \n ex: abcdefgh-0000-1a23-ab00-12a3b4cd5ef6")
+    razorpay_test_mode = fields.Boolean(string="Razorpay Test Mode", default=False, help="Turn it on when in Test Mode")
+
+    def _get_payment_terminal_selection(self):
+        return super()._get_payment_terminal_selection() + [('razorpay', 'Razorpay')]
+
+    def _call_razorpay(self, endpoint, payload=None):
+        """ Make a request to Razorpay API.
+
+        :param str endpoint: The endpoint to be reached by the request.
+        :param dict payload: The payload of the request.
+        :return The JSON-formatted content of the response.
+        :rtype: dict
+        """
+        try:
+            api_url = BASE_URL['test'] if self.razorpay_test_mode else BASE_URL['production']
+            response = requests.post(api_url+endpoint, json=payload, timeout=REQUEST_TIMEOUT)
+            response.raise_for_status()
+        except (requests.exceptions.Timeout, requests.exceptions.RequestException) as error:
+            _logger.warning("Cannot connect with Razorpay. Error: %s", error)
+            return {'error': '%s' % error}
+        res_json = response.json()
+        return res_json
+
+    def razorpay_make_payment_request(self, data):
+        body = self._razorpay_get_payment_request_body(payment_mode=True)
+        body.update({
+            'amount': data.get('amount'),
+            'externalRefNumber': data.get('referenceId')
+        })
+        response = self._call_razorpay(endpoint="pay", payload=body)
+        if response.get('success') and not response.get('errorCode'):
+            return {
+                'success': response.get('success') is True,
+                'p2pRequestId': '%s' % response.get('p2pRequestId')
+            }
+        default_error_msg = _('Razorpay payment request expected errorCode not found in the response')
+        error = response.get('errorMessage') or default_error_msg
+        return {'error': '%s' % error}
+
+    def razorpay_fetch_payment_status(self, data):
+        body = self._razorpay_get_payment_status_request_body()
+        body['origP2pRequestId'] = data.get('p2pRequestId')
+        response = self._call_razorpay(endpoint="status", payload=body)
+        if response.get('success') and not response.get('errorCode'):
+            _logger.info(response.get('message'))
+            payment_status = response.get('status')
+            if payment_status == 'AUTHORIZED' and response.get('messageCode') == 'P2P_DEVICE_TXN_DONE':
```

Comment:
Here, you assign `response.get('status')` in a variable and for `response.get('messageCode')` you compare it each time
In term of performance it is better to assign the result into a variable